### PR TITLE
[Net] Socks5 netbase back ports, less scary failures.

### PR DESCRIPTION
**Pull request by furszy**
Made a set of back ports to make Socks5 connection failures less noisy and less unnecessarily scary.
Plus added a straightforward removal of `getaddrinfo_a` which was never available at configure time.

Back ported PRs list:
#8033
#9229
#9539

### CODE REVIEW

**Review by random-zebra — APPROVED** [review]
utACK 0c1406f5c28c6ecf49b3b39a85445c53e8dce722 with a non-blocking nit.

File: src/netbase.cpp [hunk 1]
```diff
@@ -308,6 +308,21 @@ struct ProxyCredentials
     std::string password;
 };
 
+std::string Socks5ErrorString(int err)
```
[the remark this answers is not in the document]
Comment (furszy) [reply src/netbase.cpp]: Done, squashed in a1b0017 .

**Review by Fuzzbawls — APPROVED** [review]
utACK d46b3694896bb556de82fb266a9353e382eb8943

**Review by random-zebra — APPROVED** [review]
utACK d46b3694896bb556de82fb266a9353e382eb8943 and merging...

**Pull request merged**